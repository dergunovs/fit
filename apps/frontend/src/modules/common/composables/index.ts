import { ref, computed, onMounted } from 'vue';
import { useRouter, useRoute } from 'vue-router';

import LayoutAdmin from '@/common/components/LayoutAdmin.vue';
import LayoutDefault from '@/common/components/LayoutDefault.vue';

export function useLayout() {
  const router = useRouter();
  const route = useRoute();

  const isLoaded = ref(false);

  const layoutComponent = computed(() => (route.meta.layout === 'admin' ? LayoutAdmin : LayoutDefault));

  onMounted(async () => {
    await router.isReady();
    isLoaded.value = true;
  });

  return {
    isLoaded,
    layoutComponent,
  };
}

export function useRouteId(title: string, isQuery?: boolean) {
  const route = useRoute();

  const id = computed(() => {
    const value = isQuery ? route.query[title]?.toString() : route.params[title].toString();

    return value || '';
  });

  return { id };
}
