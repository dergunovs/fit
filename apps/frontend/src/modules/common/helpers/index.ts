import { toast } from 'mhz-ui';

export async function copyToClipboard(text: string, toastText: string) {
  await navigator.clipboard.writeText(text);

  toast.success(toastText);
}
