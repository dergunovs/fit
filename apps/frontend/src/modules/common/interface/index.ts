import { TPostAuthLoginDTO } from 'fitness-tracker-contracts';
import { FunctionalComponent } from 'vue';

export interface INavItem {
  url: string;
  title: string;
  icon: FunctionalComponent;
}

export interface IOnSuccess {
  create?: (id?: string) => Promise<void>;
  update?: () => Promise<void>;
  delete?: () => Promise<void>;
  login?: (user: TPostAuthLoginDTO) => Promise<void>;
  setup?: () => Promise<void>;
  register?: () => Promise<void>;
}

export interface IPromptEvent extends Event {
  prompt: () => Promise<void>;
}

export interface IGetLatestVersionDTO {
  'dist-tags': { latest: string };
}
