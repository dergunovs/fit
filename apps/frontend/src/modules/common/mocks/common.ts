import { vi } from 'vitest';

export const spyCopyToClipboard = vi.spyOn(navigator.clipboard, 'writeText').mockResolvedValue();
