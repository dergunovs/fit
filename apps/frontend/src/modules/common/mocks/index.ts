export * from './helpers';
export * from './ui';
export * from './router';
export * from './layout';
export * from './pwa';
export * from './service';
