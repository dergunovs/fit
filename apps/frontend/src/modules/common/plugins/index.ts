import { createI18n, I18nOptions } from 'vue-i18n';

import ru from '@/common/locales/ru.json';
import en from '@/common/locales/en.json';

const messages = { en, ru } as const;

const options: I18nOptions = { legacy: false, locale: 'ru', messages };

// eslint-disable-next-line @typescript-eslint/no-type-alias
export type TMessageSchema = typeof en;

export const i18n = createI18n<false, typeof options>(options);
