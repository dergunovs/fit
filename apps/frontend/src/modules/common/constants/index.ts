import IconExercise from '@/common/icons/nav-exercise.svg?component';
import IconActivity from '@/common/icons/nav-activity.svg?component';
import IconEquipment from '@/common/icons/weight.svg?component';
import IconUser from '@/common/icons/nav-user.svg?component';

import { URL_USER } from '@/user/constants';
import { URL_EXERCISE } from '@/exercise/constants';
import { URL_ACTIVITY_ADMIN } from '@/activity/constants';
import { URL_EQUIPMENT } from '@/equipment/constants';
import { INavItem } from '@/common/interface';
import { URL_SETUP } from '@/auth/constants';

export const NAV_ITEMS: INavItem[] = [
  { _id: '1', url: URL_EXERCISE, title: 'Упражнения', icon: IconExercise },
  { _id: '2', url: URL_ACTIVITY_ADMIN, title: 'Занятия', icon: IconActivity },
  { _id: '3', url: URL_EQUIPMENT, title: 'Оборудование', icon: IconEquipment },
  { _id: '4', url: URL_USER, title: 'Пользователи', icon: IconUser },
];

export const UPDATE_BUTTON_TEXT = 'Обновить';
export const CREATE_BUTTON_TEXT = 'Создать';

export const URL_HOME = '/';
export const URL_ERROR = '/404';

export const GUEST_PAGES = [URL_HOME, URL_SETUP];
