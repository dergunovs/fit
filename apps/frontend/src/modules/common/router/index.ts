import { createRouter, createWebHistory } from 'vue-router';
import { getCookieToken, logout, deleteAuthHeader } from 'mhz-helpers';

import { TOKEN_NAME } from '@/auth/constants';
import { GUEST_PAGES, URL_HOME } from '@/common/constants';
import { routes } from '@/common/router/routes';

const router = createRouter({ history: createWebHistory(), routes });

router.beforeEach((to, from, next) => {
  if (!GUEST_PAGES.includes(to.path) && !getCookieToken(TOKEN_NAME)) {
    logout(URL_HOME, deleteAuthHeader, TOKEN_NAME);
  } else {
    next();
  }
});

export { router };
