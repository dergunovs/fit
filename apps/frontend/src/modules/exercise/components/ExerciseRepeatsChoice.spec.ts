import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import ExerciseRepeatsChoice from './ExerciseRepeatsChoice.vue';

import { wrapperFactory } from '@/common/test';

const titleText = dataTest('exercise-repeats-choice-title');
const buttons = dataTest('exercise-repeats-choice-buttons');
const button = dataTest('exercise-repeats-choice-button');
const input = dataTest('exercise-repeats-choice-input');

const options = [1, 2, 3, 4, 5];
const modelValue = options[2];
const title = 'Заголовок';
const isTall = true;

let wrapper: VueWrapper<InstanceType<typeof ExerciseRepeatsChoice>>;

beforeEach(() => {
  wrapper = wrapperFactory(ExerciseRepeatsChoice, { options, modelValue, title, isTall });
});

enableAutoUnmount(afterEach);

describe('ExerciseRepeatsChoice', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(ExerciseRepeatsChoice)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows title', async () => {
    expect(wrapper.find(titleText).text()).toBe(title);
  });

  it('shows tall buttons by props', async () => {
    expect(wrapper.find(buttons).attributes('data-tall')).toBe(isTall.toString());

    await wrapper.setProps({ isTall: false });

    expect(wrapper.find(buttons).attributes('data-tall')).toBe('false');
  });

  it('shows buttons', async () => {
    expect(wrapper.findAll(button).length).toBe(options.length);
    expect(wrapper.find(button).text()).toBe(options[0].toString());
    expect(wrapper.find(button).attributes('data-current')).toBe((modelValue === options[0]).toString());
  });

  it('shows input', async () => {
    expect(wrapper.find(input).attributes('value')).toBe(modelValue.toString());
  });

  it('emits choosen value', async () => {
    await wrapper.find(button).trigger('click');

    expect(wrapper.emitted('update:modelValue')).toHaveLength(1);
    expect(wrapper.emitted()['update:modelValue'][0]).toStrictEqual([options[0]]);

    wrapper.find(input).setValue(options[1]);

    expect(wrapper.emitted('update:modelValue')).toHaveLength(2);
    expect(wrapper.emitted()['update:modelValue'][1]).toStrictEqual([options[1]]);
  });
});
