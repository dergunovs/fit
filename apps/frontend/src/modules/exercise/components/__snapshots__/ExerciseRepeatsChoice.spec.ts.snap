// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExerciseRepeatsChoice > matches snapshot 1`] = `
"<div class="_container_27a27a">
  <div data-test="exercise-repeats-choice-title">Заголовок</div>
  <div class="_repeats_27a27a" data-tall="true" data-test="exercise-repeats-choice-buttons"><button type="button" class="_button_27a27a" data-current="false" data-test="exercise-repeats-choice-button">1</button><button type="button" class="_button_27a27a" data-current="false" data-test="exercise-repeats-choice-button">2</button><button type="button" class="_button_27a27a" data-current="true" data-test="exercise-repeats-choice-button">3</button><button type="button" class="_button_27a27a" data-current="false" data-test="exercise-repeats-choice-button">4</button><button type="button" class="_button_27a27a" data-current="false" data-test="exercise-repeats-choice-button">5</button><input type="number" class="_input_27a27a" step="1" min="2" max="30" data-test="exercise-repeats-choice-input" value="3"></div>
</div>"
`;
